packages/common/src/api/tan-query/jupiter/utils.ts

```diff
@@ -0,0 +1,174 @@
+import { AudiusSdk } from '@audius/sdk'
+import {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  createAssociatedTokenAccountIdempotentInstruction,
+  createCloseAccountInstruction,
+  createTransferCheckedInstruction,
+  getAccount,
+  getAssociatedTokenAddressSync
+} from '@solana/spl-token'
+import { PublicKey, TransactionInstruction } from '@solana/web3.js'
+
+import { JUPITER_PROGRAM_ID } from './constants'
+import { UserBankManagedTokenInfo } from './types'
+
+export async function addUserBankToAtaInstructions({
+  tokenInfo,
+  userPublicKey,
+  ethAddress,
+  amountLamports,
+  sdk,
+  feePayer,
+  instructions
+}: {
+  tokenInfo: UserBankManagedTokenInfo
+  userPublicKey: PublicKey
+  ethAddress: string
+  amountLamports: bigint
+  sdk: any
+  feePayer: PublicKey
+  instructions: TransactionInstruction[]
+}): Promise<PublicKey> {
+  const mint = new PublicKey(tokenInfo.mintAddress)
+  const ata = getAssociatedTokenAddressSync(mint, userPublicKey, true)
+
+  try {
+    await getAccount(sdk.services.solanaClient.connection, ata)
+  } catch (e) {
+    instructions.push(
+      createAssociatedTokenAccountIdempotentInstruction(
+        feePayer,
+        ata,
+        userPublicKey,
+        mint
+      )
+    )
+  }
+
+  const secpTransferInstruction =
+    await sdk.services.claimableTokensClient.createTransferSecpInstruction({
+      amount: amountLamports,
+      ethWallet: ethAddress,
+      mint: tokenInfo.claimableTokenMint,
+      destination: ata,
+      instructionIndex: instructions.length
+    })
+  const transferInstruction =
+    await sdk.services.claimableTokensClient.createTransferInstruction({
+      ethWallet: ethAddress,
+      mint: tokenInfo.claimableTokenMint,
+      destination: ata
+    })
+
+  instructions.push(secpTransferInstruction, transferInstruction)
+  return ata
+}
+
+export async function addAtaToUserBankInstructions({
+  tokenInfo,
+  userPublicKey,
+  ethAddress,
+  amountLamports,
+  sourceAta,
+  sdk,
+  feePayer,
+  instructions
+}: {
+  tokenInfo: UserBankManagedTokenInfo
+  userPublicKey: PublicKey
+  ethAddress: string
+  amountLamports: bigint
+  sourceAta: PublicKey
+  sdk: AudiusSdk
+  feePayer: PublicKey
+  instructions: TransactionInstruction[]
+}): Promise<PublicKey> {
+  const mint = new PublicKey(tokenInfo.mintAddress)
+  const userBankAddress =
+    await sdk.services.claimableTokensClient.deriveUserBank({
+      ethWallet: ethAddress,
+      mint: tokenInfo.claimableTokenMint
+    })
+
+  instructions.push(
+    createTransferCheckedInstruction(
+      sourceAta,
+      mint,
+      userBankAddress,
+      userPublicKey,
+      amountLamports,
+      tokenInfo.decimals
+    ),
+    createCloseAccountInstruction(sourceAta, feePayer, userPublicKey)
+  )
+  return userBankAddress
+}
+
+export function updateJupiterAtaCreationFeePayer(
```

Comment:
afaik the only way the ATA gets created from jupiter is via the setup instructions, which we aren't using